addons/stock_dropshipping/models/purchase.py

```diff
@@ -17,7 +17,9 @@ def _prepare_stock_moves(self, picking):
         return res
 
     def _merge_in_existing_line(self, product_id, product_qty, product_uom, location_id, name, origin, values):
-        if values.get('route_ids') and values['route_ids'] == self.env.ref('stock_dropshipping.route_drop_shipping'):
+        if self.sale_line_id and (self.sale_line_id.id != values.get('sale_line_id', 0)):
```

Comment:
`, 0` is probably not necessary since:

```
>>> 1 != None
True
```

I guess `(self.sale_line_id.id != values.get('sale_line_id', 0))` is to take into account the modification of a validated SO, right?

Reply:
you probably won't merge a dropship in an existing line without dropship. I would use
``` python3
values.get('sale_line_id') and values['sale_line_id'] != self.sale_line_id.id
```

Reply:
The zero was just to check an integer for an integer, but it is useless indeed.
Your guess is right; the test you added checked that the line value was increased, which was not the case since without that check it would have created another line.

I went for Arnold's version then.